controllers/provider-alicloud/pkg/cmd/config.go

```diff
@@ -77,8 +77,13 @@ func (c *Config) ApplyMachineImages(machineImages *[]config.MachineImage) {
 }
 
 // ApplyMachineImageOwnerSecretRef sets the given machine image owener secret reference to those of this Config.
-func (c *Config) ApplyMachineImageOwnerSecretRef(secretRef *corev1.SecretReference) {
-	*secretRef = *c.Config.MachineImageOwnerSecretRef
+func (c *Config) ApplyMachineImageOwnerSecretRef(secretRef **corev1.SecretReference) {
```

Comment:
> Can't we use
> 
> and
> 
> ```go
> 		*secretRef = corev1.SecretReference{
> 			Name:      c.Config.MachineImageOwnerSecretRef.Name,
> 			Namespace: c.Config.MachineImageOwnerSecretRef.Namespace,
> 		}
> ```

`MachineImageOwnerSecretRef` itself is a `*corev1.SecretReference`, so that it can be checked by `if MachineImageOwnerSecretRef == nil`. If we make it a `corev1.SecretReference` , `&MachineImageOwnerSecretRef` will never be nil as it is `&{}`. WDYT?

Reply:
Sorry, my bad.